tiatoolbox/utils/image.py

```diff
@@ -281,7 +294,7 @@ def sub_pixel_read(
         image = np.array(image)
     bounds = np.array(bounds)
     _, bounds_size = bounds2locsize(bounds)
-    if np.size(stride) == 2:
+    if np.size(stride) == 1:
         stride = np.tile(stride, 2)
```

Comment:
Why? Did you see bug in previous logic here? You may want to tag John.

Reply:
Yes I checked this.